src/db/postgres.ts

```
@@ -0,0 +1,40 @@
+import pg from 'pg';
+
+
+class Postgres {
+    constructor(private db: DBType) {
+      this.init();
+    }
+  
+    async init() {
+        this.getPool().connect(function(err, client, done) {
```

It would be great to use async (check the postgres docs, use await) instead of a callback (connect(function(...)).
Then simply let it throw, so we can ensure that the connection was successful and the table was created before our application starts handling requests.
In instances we would do const postgres = await new Postgres({...}).init()
The init function would have to return `this` if it didn't throw.
The constructor would no longer call this.init() as the constructor can't use async await.

PS: This is optional, do it only if you feel like and have the time.